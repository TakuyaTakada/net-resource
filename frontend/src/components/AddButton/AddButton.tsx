import React from "react";
import { Fab } from "@material-ui/core";
import AddIcon from "@material-ui/icons/Add";
import useStyles from "./Style";

type PropsType = {
  handleCreateOpen(): void;
};

const AddButton: React.FC<PropsType> = ({ handleCreateOpen }) => {
  const classes = useStyles();
  return (
    <Fab
      color="primary"
      aria-label="add"
      className={classes.fab}
      size="small"
      onClick={handleCreateOpen}
    >
      <AddIcon />
    </Fab>
  );
};

export default AddButton;
