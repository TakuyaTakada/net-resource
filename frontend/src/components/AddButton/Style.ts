import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    fab: {
      position: "absolute",
      bottom: theme.spacing(2),
      right: "10%"
    }
  })
);

export default useStyles;
