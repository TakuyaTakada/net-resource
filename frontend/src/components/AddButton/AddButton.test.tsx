import React from "react";
import { render, cleanup } from "@testing-library/react";
import AddButton from "./AddButton";

afterEach(cleanup);

describe("AddButton", () => {
  test("Snapshot", () => {
    const { asFragment } = render(<AddButton handleCreateOpen={() => null} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
