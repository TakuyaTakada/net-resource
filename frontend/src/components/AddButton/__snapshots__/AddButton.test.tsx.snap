// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButton Snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="add"
    class="MuiButtonBase-root MuiFab-root makeStyles-fab-1 MuiFab-sizeSmall MuiFab-primary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiFab-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
